proto/tripcommon.proto

```diff
@@ -85,6 +85,8 @@ message Location {
   repeated PathEdge filtered_edges = 23;
   optional uint32 original_index = 24;
   optional LatLng projected_ll = 25;
+  optional uint32 leg_shape_index = 35;
+  optional double distance_from_origin = 36;
```

Comment:
nit: technically its from origin of the leg so it might make sense to name it accordingly. `distance_within_leg` or `distance_from_leg_origin`? just to avoid confusion for people who are trying to figure out what it is without reverse engineering the rest of the code